opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -113,10 +113,30 @@ class BatchExportSpanProcessor(SpanProcessor):
     def __init__(
         self,
         span_exporter: SpanExporter,
-        max_queue_size: int = 2048,
-        schedule_delay_millis: float = 5000,
-        max_export_batch_size: int = 512,
+        max_queue_size: int = None,
+        schedule_delay_millis: float = None,
+        max_export_batch_size: int = None,
+        export_timeout_millis: float = None,
     ):
+
+        if max_queue_size is None:
+            max_queue_size = Configuration().get("BSP_MAX_QUEUE_SIZE", 2048)
+
+        if schedule_delay_millis is None:
+            schedule_delay_millis = Configuration().get(
+                "BSP_SCHEDULE_DELAY_MILLIS", 5000
+            )
+
+        if max_export_batch_size is None:
+            max_export_batch_size = Configuration().get(
```

Comment:
We should have validation to ensure less than or equal to max queue size.

Reply:
I think this is already implemented [here](https://github.com/open-telemetry/opentelemetry-python/pull/1120/files/aa49ad5a7c469caa0e01543d7f84aec87e9535df#diff-af7e6beb87f8ae3bf2cb9280ca2b6032R151).